### PR TITLE
fix(coordinator): update permission control to set active verifiers to only provers

### PR DESCRIPTION
**Pull request by eguajardo**
## Description

https://axelarnetwork.atlassian.net/browse/AXE-4365

## Todos

- [ ] Unit tests
- [ ] Manual tests
- [ ] Documentation
- [ ] Connect epics/issues

### CODE REVIEW

File: contracts/coordinator/src/state.rs [hunk 1]
```diff
@@ -1,18 +1,86 @@
...
+}
+
+const CHAIN_PROVER_INDEXED_MAP: IndexedMap<
+    (ChainName, ProverAddress),
+    ChainProverRecord,
+    ChainProverIndexes,
+> = IndexedMap::new(
+    "chain_prover_map",
+    ChainProverIndexes {
+        by_chain: UniqueIndex::new(|d| d.chain.clone(), "chain_prover_map_by_chain"),
```
Comment (cjcobb23) [inline contracts/coordinator/src/state.rs:38]: why do we need this index? The underlying map uses chain as a primary key, so can't we just look up directly?

File: contracts/coordinator/src/state.rs [hunk 2]
```diff
@@ -1,18 +1,86 @@
...
+}
+
+pub fn save_prover_for_chain(
+    storage: &mut dyn Storage,
+    chain: ChainName,
+    prover: ProverAddress,
+) -> Result<(), ContractError> {
+    CHAIN_PROVER_INDEXED_MAP.save(
+        storage,
+        (chain.clone(), prover.clone()),
```
Comment (cjcobb23) [inline contracts/coordinator/src/state.rs:75]: why does the prover need to be part of the primary key, if its part of the value itself?

File: contracts/coordinator/src/state.rs [hunk 3]
```diff
@@ -23,11 +73,11 @@ impl<'a> IndexList<VerifierProverRecord> for VerifierSetIndex<'a> {
 #[cw_serde]
 pub struct VerifierProverRecord {
     pub prover: ProverAddress,
-    pub verifier: Addr,
+    pub verifier: VerifierAddress,
 }
 
 pub const VERIFIER_PROVER_INDEXED_MAP: IndexedMap<
```
[the remark this answers is not in the document]
Comment (haiyizxx) [reply contracts/coordinator/src/state.rs:79]: agree with CJ, seems can be combined into one

**Review by cjcobb23 — APPROVED** [review]

File: contracts/coordinator/src/state.rs (hunk 3 above)
[the remark this answers is not in the document]
Comment (eguajardo) [reply contracts/coordinator/src/state.rs:67]: been back and forth discussing this and can't find a way to combine them. Main issue is that the verifier address needs to be part of the primary key, but when registering a new prover we don't have any verifier address yet.

**Pull request merged**